### PR TITLE
[prometheus-adapter] Add sidecar container support

### LINKED ISSUE #3739: [prometheus-adapter] Add Support for sidecar containers
### Is your feature request related to a problem ?

Currently there is no support for having sidecar containers in the helm chart. If a service like AWS Managed Prometheus (AMP) is being used then the adapter needs support for authentication with AWS using sigv4. Creating this issue because https://github.com/prometheus-community/helm-charts/issues/1568 is closed because of being stale.

### Describe the solution you'd like.

It would be good if there was an option of defining sidecar containers within the helm chart. Something like a variable inside the values file, which can be referenced in the deployment template.

### Describe alternatives you've considered.

To get past this issue right now, I've forked the helm chart and added the required changes in the fork.

### Additional context.

_No response_

### PR DESCRIPTION
**Pull request by rpriyanshu9**
#### What this PR does / why we need it

This PR makes it possible to add sidecar containers to a prometheus-adapter deployment. This lets you put e.g. a sigv4 proxy sidecar (https://github.com/awslabs/aws-sigv4-proxy) alongside the main container, which we need in order to connect to Amazon Managed Service for Prometheus’s query endpoint.

#### Which issue this PR fixes

- fixes #3739

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

**Comment by rpriyanshu9** [comment]
👋🏼 @MattiasGees @hectorj2f @steven-sheehy Could you please take a look?

**Comment by rpriyanshu9** [comment]
PTAL @hectorj2f @MattiasGees @steven-sheehy

File: charts/prometheus-adapter/Chart.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 apiVersion: v1
 name: prometheus-adapter
-version: 4.4.1
+version: 4.4.2
```
Comment (steven-sheehy) [inline charts/prometheus-adapter/Chart.yaml:3]:
```suggestion
version: 4.6.0
```

File: charts/prometheus-adapter/values.yaml [hunk 2]
```diff
@@ -271,3 +271,6 @@ certManager:
   enabled: false
   caCertDuration: 43800h
   certDuration: 8760h
+
+# Sidecar containers that run alongside prometheus-adapter container.
+sideCarContainers: []
```
Comment (steven-sheehy) [inline charts/prometheus-adapter/values.yaml:276]: `extraContainers` is more widely used in this repository. Also, move it above `extraVolumes` so it's more sorted.

```suggestion
# Additional containers to add to the pod
extraContainers: []
```

**Review by steven-sheehy — CHANGES_REQUESTED** [review]

**Comment by rpriyanshu9** [comment]
@steven-sheehy I've implemented the requested changes. Could you please take a look?

**Review by steven-sheehy — APPROVED** [review]
LGTM. Thanks for your contribution

**Pull request merged**